app/javascript/src/Applications/create_application.js

```diff
@@ -47,6 +47,14 @@ export class CreateApplication {
     return this.relationPlansServices[this.selectedPlan]
   }
 
+  get createServicePlanPath (): string {
+    return this.metadata.data('create_service_plan_path')
+  }
+
+  getCreateServicePlanPathForService (serviceId: number): string {
+    return this.createServicePlanPath.replace(':service_id', String(serviceId))
```

Comment:
Why not directly this?
```javascript
getCreateServicePlanPathForService (serviceId: number): string {
  return this.metadata.data('create_service_plan_path').replace(':service_id', String(serviceId))
}
```
Then you wouldn't need the previous method.

I ask because in this same file, a bit above, you have this:
https://github.com/3scale/porta/blob/ec754ae33e58eae72c969180e9f58b64c6289f3f/app/javascript/src/Applications/create_application.js#L38-L40

So just for the sake of following always the same pattern inside the file 🤔 Besides that url with `:service_id` actually does not point to any valid place.
But this is 100% up to the JS team to decide 😄